scripts/g.extension/testsuite/test_addons_download.py

```diff
@@ -0,0 +1,141 @@
+"""
+TEST:      test_addons_download.py
+
+AUTHOR(S): Stefan Blumentrath <stefan dot blumentrath at nina dot no)
+           based on test_addons_modules.py by
+           Vaclav Petras <wenzeslaus gmail com>
+
+PURPOSE:   Test for g.extension individual modules/extensions download
+
+COPYRIGHT: (C) 2020 Stefan Blumentrath, Vaclav Petras,
+           and by the GRASS Development Team
+
+           This program is free software under the GNU General Public
+           License (>=v2). Read the file COPYING that comes with GRASS
+           for details.
+"""
+
+import sys
+import unittest
+
+from pathlib import Path
+
+from grass.gunittest.case import TestCase
+from grass.gunittest.main import test
+from grass.gunittest.utils import silent_rmtree
+
+ms_windows = sys.platform == "win32"
+
+
+class TestModuleDownloadFromDifferentSources(TestCase):
+
+    # MS Windows install function requires absolute paths
+    install_prefix = Path("gextension_test_install_path").absolute()
+
+    files = [
+        install_prefix.joinpath("scripts", "r.example.plus"),
+        install_prefix.joinpath("docs", "html", "r.example.plus.html"),
+    ]
+
+    def setUp(self):
+        """Make sure we are not dealing with some old files"""
+        if self.install_prefix.exists():
+            files = list(path.name for path in self.install_prefix.iterdir())
+            if files:
+                RuntimeError(
+                    "Install prefix path '{}' contains files {}".format(
+                        str(self.install_prefix), files
+                    )
+                )
+
+    def tearDown(self):
+        """Remove created files"""
+        silent_rmtree(str(self.install_prefix))
```

Comment:
Out of scope for this PR, but silent_rmtree should support Path (same for all path handling functions). I recall something about shutil not supporting it initially, but later on adding it.